test/functional/page_objects/settings_page.js

```diff
@@ -328,6 +329,17 @@ export function SettingsPageProvider({ getService, getPageObjects }) {
       return await this.getIndexPatternIdFromUrl();
     }
 
+    //adding a method to check if the create index pattern button is visible(while adding more than 1 index pattern)
+
+    async clickOptionalAddNewButton() {
+      const buttonParent = await testSubjects.find('createIndexPatternParent');
```

Comment:
Hmm. I wonder if you can just do:
```
const buttonVisible = await testSubjects.find('createIndexPatternButton');
```

and have it work the same?

Reply:
@chrisronline the issue here is that the `Create Index Pattern` button only exists where there is already a default index pattern set.  So one option is to do a `testSubjects.find` and catch and throw away if that fails.  But that will always take the findTimeout which I think is 10 seconds :-(
By finding the parent element which is always there, we can check if it contains the button and it's very fast for both cases.

Reply:
Yea that's fair, but will this test fail if the button doesn't exist? We just don't do anything if the button does not exist. Does that mean the test calling this will then handle that somehow?

Reply:
I think there's either the button (and not the index pattern name entry field)
or there's no button (and there is the index pattern name entry field) (unless there's a 3rd case I'm not thinking of).
So the alternate way would be to see if the entry field is there, and if it's not, then click the button that should be there.
In either case, we'd prefer to not have the 10s find timeout when something's not there.

But, this code isn't explicitly checking both conditions.  So if the button isn't there, it will try to find the field to enter the index pattern name.  And if that's not there it will fail.  But it should fail in that case.